src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -1,62 +1,78 @@
+            if (html.indexOf(suffix) < 0) {
+                e.innerHTML = e.innerHTML.replace(/.+/, '$&' + suffix) // insert before EOL
+            }
         }
-    }
-    var nodes = $$('.pipeline-new-node')
-    var enclosings = new Map() // id → enclosingId
-    var labels = new Map() // id → label
-    for (var i = 0; i < nodes.length; i++) {
-        var node = nodes[i]
-        var oid = node.getAttribute('nodeId')
-        enclosings.set(oid, node.getAttribute('enclosingId'))
-        labels.set(oid, node.getAttribute('label'))
-    }
-    var id = nodeId
-    while (true) {
-        id = enclosings.get(id)
-        if (id == null) {
-            break
+
+        // generate hide/show hyperlinks
+        if (startId == null || startId == nodeId) {
+            e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
         }
-        var label = labels.get(id)
-        if (label != null && label.indexOf('Branch: ') == 0) {
-            var branch = label.substring(8)
-            var ss = document.styleSheets[0]
-            // TODO https://stackoverflow.com/a/18990994/12916 does not scale well to add width: 25em; text-align: right
-            ss.insertRule('.pipeline-node-' + nodeId + '::before {content: "[' + branch.escapeHTML() + '] "; color: #9A9999; position: absolute; transform: translateX(-100%)}', ss.cssRules == null ? 0 : ss.cssRules.length)
-            break
+
+        // The CSS rule for branch names only needs to be added once per node, so we
+        // check in case we are viewing the truncated log and have already processed
+        // a duplicate synthetic span element for this node.
+        var maybeDupeNodes = $$('[nodeid=\"' + nodeId + '\"].pipeline-new-node');
+        for (var i = 0; i < maybeDupeNodes.length; i++) {
+            var node = maybeDupeNodes[i];
+            if (node !== e && node.processedNewNodeConsoleNote) {
+                return;
+            }
         }
-    }
+
+        enclosings.set(oid, e.getAttribute('enclosingId'))
+        labels.set(oid, e.getAttribute('label'))
+
+        var id = nodeId
+        while (true) {
+            id = enclosings.get(id)
+            if (id == null) {
+                break
+            }
+            var label = labels.get(id)
+            if (label != null && label.indexOf('Branch: ') == 0) {
+                var branch = label.substring(8)
+                var ss = document.styleSheets[0]
+                // TODO https://stackoverflow.com/a/18990994/12916 does not scale well to add width: 25em; text-align: right
+                ss.insertRule('.pipeline-node-' + nodeId + '::before {content: "[' + branch.escapeHTML() + '] "; color: #9A9999; position: absolute; transform: translateX(-100%)}', ss.cssRules == null ? 0 : ss.cssRules.length)
+                break
+            }
+        }
+    }, 5000)
```

Comment:
5 seconds is a long time to wait. Is 1 second (or maybe even less) not good enough just to get it out of the critical path for the main page load?